server/src/main/java/org/opensearch/index/shard/CloseableRetryableRefreshListener.java

```diff
@@ -26,11 +28,9 @@
  */
 public abstract class CloseableRetryableRefreshListener implements ReferenceManager.RefreshListener, Closeable {
 
-    /**
-     * Total permits = 1 ensures that there is only single instance of performAfterRefresh that is running at a time.
-     * In case there are use cases where concurrency is required, the total permit variable can be put inside the ctor.
-     */
-    private static final int TOTAL_PERMITS = 1;
+    private static final int TOTAL_PERMITS = Integer.MAX_VALUE;
```

Comment:
With the value > 1, is it possible to run a retry and new aftreRefresh concurrently?

Reply:
Yes, that's possible.

Reply:
Will it create any issue due to concurrent runs? Is it expected from the implementing listener to handle such cases?

Reply:
Made the changes as discussed. We just need 1 permit.